iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/Tools.java

```diff
@@ -39,21 +50,47 @@ public static String retrieveEnvironmentVariableValue(String environmentVariable
 
     public static String retrieveEnvironmentVariableValue(String environmentVariableName, String defaultValue)
     {
-        try
+        String environmentVariableValue;
+        if (ANDROID_ENV_VAR.containsKey(environmentVariableName))
+        {
+            environmentVariableValue = ANDROID_ENV_VAR.get(environmentVariableName);
+        }
+        else
         {
-            String value = retrieveEnvironmentVariableValue(environmentVariableName);
+            environmentVariableValue = System.getenv().get(environmentVariableName);
+        }
 
-            if (value == null || value.isEmpty())
-            {
-                return defaultValue;
-            }
+        if (environmentVariableValue == null || environmentVariableValue.isEmpty())
+        {
+            return defaultValue;
+        }
 
-            return value;
+        return environmentVariableValue;
+    }
+
+    private static Map<String, String> retrieveAndroidEnvVariables()
+    {
+        Map<String, String> envVariables = new HashMap<>();
+        try
+        {
+            Class buildConfig = Class.forName(ANDROID_BUILD_CONFIG_CLASS);
+            Arrays.stream(buildConfig.getFields()).forEach(field -> {
+                try
+                {
+                    envVariables.put(field.getName(), field.get(null).toString());
+                }
+                catch (IllegalAccessException e)
+                {
+                    // disregard
```

Comment:
we should log something here, to indicate that env var retrieval from buildConfig failed.

Reply:
Our e2e test suite currently doesn't log anything but test failures

Reply:
I can see about adding this, though